src/CoherenceBuild/CIVolatileFeedPublisher.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet;
+
+namespace CoherenceBuild
+{
+    public static class CIVolatileFeedPublisher
+    {
+        public static void CleanupVolatileFeed(
+            ProcessResult processResult, 
+            string volatileShare)
+        {
+            var expandedPackageRepository = new ExpandedPackageRepository(new PhysicalFileSystem(volatileShare));
+            var packagesToDelete = new List<IPackage>();
+            foreach (var package in expandedPackageRepository.GetPackages())
+            {
+                List<PackageInfo> coherentPackageInfos;
+                PackageInfo coherentPackageInfo;
+                if (processResult.CoreCLRPackages.TryGetValue(package.Id, out coherentPackageInfos))
+                {
+                    if (coherentPackageInfos.Any(c => c.Package.Version <= package.Version))
+                    {
+                        // Allow packages in the volatile share to be higer than the version that is coherent.
+                        // When we have multiple CoreCLR versions, only delete it if it is lower than all packages
+                        // for that id.
+                        continue;
+                    }
+                }
+                else if (processResult.ProductPackages.TryGetValue(package.Id, out coherentPackageInfo))
+                {
+                    if (coherentPackageInfo.Package.Version <= package.Version)
```

Comment:
The idea is to allow newer versions of packages to be present in the volatile feed.